crates/asm/common/src/aux.rs

```diff
@@ -1,56 +1,87 @@
-use borsh::{BorshDeserialize, BorshSerialize};
+use std::{any::Any, collections::BTreeMap};
 
-use crate::{AsmError, Subprotocol};
+use bitcoin::{BlockHash, Txid, hashes::Hash};
+use borsh::{
+    BorshDeserialize, BorshSerialize,
+    io::{Read, Write},
+};
+use strata_l1_txfmt::SubprotocolId;
+use strata_mmr::MerkleProof;
 
-/// A single auxiliary input request from a subprotocol during preprocessing.
-///
-/// The `data` field contains the raw bytes that will be processed to generate
-/// the corresponding auxiliary input data in the final [`AuxPayload`].
-#[derive(Debug)]
-pub struct AuxRequest {
-    /// Raw data for the auxiliary input request.
-    data: Vec<u8>,
+use crate::{AsmLogEntry, L1TxIndex};
+
+/// Table mapping subprotocol IDs to their corresponding auxiliary payloads.
+pub type AuxDataTable = BTreeMap<SubprotocolId, AuxInput>;
+
+/// Compact representation of an MMR proof for an ASM log leaf.
+pub type LogMmrProof = MerkleProof<[u8; 32]>;
+
+/// Supported auxiliary queries that subprotocols can register during preprocessing.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum AuxRequestSpec {
+    /// Request logs emitted across a contiguous range of L1 blocks (inclusive).
+    AsmLogQueries { start_block: u64, end_block: u64 },
+    /// Request the full deposit request transaction referenced by the deposit.
+    DepositRequestTx { txid: Txid },
 }
 
-impl AuxRequest {
-    pub fn new(data: Vec<u8>) -> Self {
-        Self { data }
-    }
+/// Responsible for recording auxiliary requests emitted during preprocessing.
+pub trait AuxRequestCollector: Any {
+    /// Records that the transaction at `tx_index` requires auxiliary data described by `request`.
+    fn request_aux_input(&mut self, tx_index: L1TxIndex, request: AuxRequestSpec);
 
-    pub fn data(&self) -> &[u8] {
-        &self.data
-    }
+    /// Exposes the collector as a `&mut dyn Any` for downcasting.
+    fn as_mut_any(&mut self) -> &mut dyn Any;
```

Comment:
Definitely just replace this with a container for individual requests.  It should be like the converse to the `AuxData` (just not bound to txs, the only thing we need binding to would be the subprotocol ID).  It's just adding indirection like this, we don't get anything out of it being generic.